### PR TITLE
feat: create hover view

### LINKED ISSUE #828: feat: Hovered courses Appear on the Calendar
chrome-capture-2023-11-1

The feature should resemble Coursicle's implementation, with a semi-transparent state. Also, shoutout Ryan for suggesting this up like six months ago.

### PR DESCRIPTION
**Pull request by KevinWu098**
## ~~WIP~~ To-Dos
- [x] Show all meetings of the section
- [x] Handle hovered sections that are already added
- [x] Disable in added pane (possibly solved by task 2)
- [x] Make not an eyesore

## Summary
1. Implement a `HoveredStore` to store the currently hovered course
2. `SectionTableBody` will access and update `HoveredStore` when a section is hovered
3. Hovered Courses will be appended to events passed to the Calendar

chrome-capture-2023-11-6 (1)

## Test Plan
1. Check that it correctly displays the course on the calendar
2. Light and Dark mode compatibility
3. Doesn't activate on Added Pane
4. Doesn't activate on already added courses

## Issues
Closes #828 

## Future Followup
1. Remove the schedule conflict feature, since this covers much of the same functionality? Unsure on this (since SC is my baby 🥺, but it's a valid question)

### CODE REVIEW

**Review by EricPedley — COMMENTED** [review]
Code looks fine, but I think the classes instantly showing up looks a little jarring. I think we should add a delay before the class shows up previewed on the calendar and maybe an animation to fade it in. We can bring this up in the meeting tomorrow to get more input on what to do from a UI/UX perspective here.

**Comment by KevinWu098** [comment]
TL;DR of latest changes:

1. "Hovered" courses are now 80% opacity
2. Settings Menu has some updated styling
3. Preview Mode is now in settings menu, defaulting false

I've chosen not to move Schedule Conflict into the experimental menu, as after some further thought, I agree with Eric on that it's a feature people have gotten used to, but I'm happy to circle back to that as needed

**Review by EricPedley — CHANGES_REQUESTED** [review]
Ok one more small change: can we call the toggle in the settings menu "Hover to Preview" instead of "Preview Mode"? I think it makes it more clear what it does. LMK if you agree or not. After we decide this, good to merge.

**Review by EricPedley — APPROVED** [review]
